Support smaller validator sets (#514)
gavofyork/graypaper: text/reporting_assurance.tex

\begin{align}
  \label{eq:availableworkreports}
  \justbecameavailable &\equiv \sq{\build{
      \reportspostjudgement\subb{\cX}_\rs¬workreport
    }{
      \cX \orderedin \coreindex,\;
      \sum_{a \in \xtassurances}\!a_\xa¬availabilities\subb{\cX}\,>\,\twothirds\,\len{\activeset}
    }}
\end{align}

This value is utilized in the definition of both $\accountspost$ and $\reportspostguarantees$ which we will define presently as equivalent to $\reportspostjudgement$ except for the removal of items which are either now available or have timed out:
\begin{equation}
  \label{eq:reportspostguaranteesdef}
  \begin{aligned}
    \forall \cX \in \coreindex: \reportspostguarantees\subb{\cX} \equiv {} &\begin{cases}
      \none &\when p = \none \vee p_\rs¬workreport \in \justbecameavailable \vee {} \\
      &\phantom{\when} \H_\¬timeslot \ge p_\rs¬timestamp + \Cassurancetimeoutperiod \vee \len{\activeset} \ne \len{\activeset'} \\
      p &\otherwise
    \end{cases} \\
    &\where p = \reportspostjudgement\subb{\cX}
  \end{aligned}
\end{equation}

> Note that work-reports may time out earlier than usual if the size of the active validator key set $\activeset$ changes.

Reviewer: ```suggestion
Note that all pending work-reports are cleared if the size of the active validator key set $\activeset$ changes.
```

Just personal preference, feel free to ignore :-)

  \forall g &\in \xtguarantees : g_\xg¬credential = \sqorderuniqby{v}{\tup{v, s} \in g_\xg¬credential}
\end{align}

The signature must be one whose public key is that of the validator identified in the credential, and whose message is the serialization of the hash of the work-report. The signing validators must have been assigned to the core in the given timeslot $\xg¬timeslot$, which must either be in the same rotation as this block's timeslot or in the previous rotation. Use of an inactive core is not permitted even if a timeslot in the previous rotation is used and the core was active then. Formally:
\begin{gather}
  \label{eq:guarantorsig}
  \begin{aligned}
    &\begin{aligned}
      \forall \tup{\xg¬workreport, \xg¬timeslot, \xg¬credential} &\in \xtguarantees,\\
      \forall \tup{v, s} &\in \xg¬credential
    \end{aligned} : \abracegroup[\,]{

>       &v < \len{\mathbf{k}} \wedge \mathbf{c}\sub{v} = \wrX_\wr¬core < \frac{\len{\activeset'}}{3}\\

Reviewer: ```suggestion
      &v < \len{\mathbf{k}} \wedge \mathbf{c}\subb{v} = \wrX_\wr¬core < \frac{\len{\activeset'}}{3}\\
```

Subscript notation for sequences

Author: Subscript notation is permitted for sequences, from the notation section:

> Sequences are subscriptable, thus a specific item at index i within a sequence s may be denoted s[i], or where unambiguous, s_i

It is used in a few places, presumably because it's a bit shorter. I think it's fine here?
